src/test/java/core/basesyntax/service/WarehouseServiceImplTest.java

```diff
@@ -0,0 +1,104 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.ProductDao;
+import core.basesyntax.dao.ProductDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.operations.BalanceOperation;
+import core.basesyntax.operations.Operation;
+import core.basesyntax.operations.Operations;
+import core.basesyntax.operations.PurchaseOperation;
+import core.basesyntax.operations.ReturnOperation;
+import core.basesyntax.operations.SupplyOperation;
+import core.basesyntax.strategy.OperationsStrategy;
+import core.basesyntax.strategy.OperationsStrategyImpl;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class WarehouseServiceImplTest {
+    private static final String INPUT = "fruit,quantity" + System.lineSeparator()
+            + "banana,152" + System.lineSeparator()
+            + "apple,90";
+    private static WarehouseService warehouseService;
+    private static ProductDao productDao;
+    private static List<String> correctDataFromFile;
+    private static List<String> dataWithWrongOperation;
+    private static List<String> dataWithWrongAmount;
+    private static Map<Product, Integer> expectedMap;
+
+    @BeforeClass
```

Comment:
Shouldn't it be before each test ?

Reply:
No, it contains only Objects that have to be created once. I don't change them in tests.